apiserver/authentication/jwt/jwt.go

```diff
@@ -50,32 +62,22 @@ type TokenEntity struct {
 	User names.UserTag
 }
 
-// TokenParser parses a jwt token returning the token and
-// entity derived from the token subject.
-type TokenParser interface {
-	// Parse parses the supplied token string and returns both the constructed
-	// jwt and the entity found within the token.
-	Parse(ctx context.Context, tok string) (jwt.Token, authentication.Entity, error)
-}
-
-func NewAuthenticator(refreshURL string) *JWTAuthenticator {
-	httpClient := &http.Client{
-		Timeout: 30 * time.Second,
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse
-		},
-	}
-	return NewAuthenticatorWithHTTPClient(httpClient, refreshURL)
-}
-
-func NewAuthenticatorWithHTTPClient(
-	client *http.Client,
-	refreshURL string,
-) *JWTAuthenticator {
-	return &JWTAuthenticator{
-		httpClient: client,
-		refreshURL: refreshURL,
+// Authenticate implements EntityAuthenticator
+func (j *JWTAuthenticator) Parse(ctx context.Context, tok string) (jwt.Token, TokenEntity, error) {
+	token, err := j.parser.Parse(ctx, tok)
+	if err != nil {
+		// Return a not implemented error if we cannot parse tokens
```

Comment:
```suggestion
		// Return a not provisioned error if we cannot parse tokens
```

Reply:
The current comment is correct, if parser is not configured we return a NotImplemented error.
I've reworded the comment to `Return a not implemented error if the parser is not configured.`